resources/scponlyc_user.rb

```diff
@@ -0,0 +1,55 @@
+# To learn more about Custom Resources, see https://docs.chef.io/custom_resources.html
+
+resource_name :scponlyc_user
+
+default_action :create
+
+property :pub_dir, String, default: '/pub'
+property :public_key, String
+property :private_key, String
+
+action :create do
+  run_context.include_recipe 'osl-scponly::default'
+
+  directory new_resource.pub_dir do
+    owner 'root'
+    group 'scponly'
+    mode '0770'
+    recursive true
+  end
+
+  user new_resource.name do
+    gid 'scponly'
+    home "/home/#{new_resource.name}"
+    manage_home true
+    shell '/usr/sbin/scponlyc'
+  end
+
+  group new_resource.name do
+    members new_resource.name
+  end
+
+  directory "/home/#{new_resource.name}" do
+    mode '0500'
+  end
+
+  directory "/home/#{new_resource.name}/.ssh" do
+    mode '0500'
+    owner new_resource.name
+    group new_resource.name
+  end
+
+  file "/home/#{new_resource.name}/.ssh/authorized_keys" do
+    content new_resource.public_key
+    mode '0400'
+    owner new_resource.name
+    group new_resource.name
+  end
+
+  cookbook_file "/home/#{new_resource.name}/.ssh/#{new_resource.private_key}" do
```

Comment:
This should be using the file resource like @khashf used in his proj-phpbb PR.

Reply:
Didn't I? Looking at @khashf PR, we're handling the private key the exact same way.

Reply:
Out of curiosity why would file be better? Private key format can be so sensitive it seems easier to just copy over a key file rather than define an attribute.

Reply:
In practice this wouldn't be set with an attribute. It'd likely be pulled from an encrypted data bag and set as a parameter to this resource. Using a file resource instead of a cookbook_file makes it so that you don't have to create a file in the cookbook. It can just be a parameter.

Reply:
ah alright that makes sense.